pkg/apiserver/apiserver_test.go

```diff
@@ -94,11 +96,10 @@ func init() {
 	// api.Status is returned in errors
 
 	// "internal" version
-	api.Scheme.AddKnownTypes("", &Simple{}, &SimpleList{},
-		&api.Status{})
+	api.Scheme.AddKnownTypes("", &Simple{}, &SimpleList{}, &api.Status{}, &api.ListOptions{})
 	// "version" version
 	// TODO: Use versioned api objects?
-	api.Scheme.AddKnownTypes(testVersion, &Simple{}, &SimpleList{}, &api.DeleteOptions{}, &api.Status{})
+	api.Scheme.AddKnownTypes(testVersion, &Simple{}, &SimpleList{}, &v1beta1.DeleteOptions{}, &v1beta1.Status{}, &v1beta1.ListOptions{})
```

Comment:
Yes, the apiserver_test package needs to define a "version2" that uses v1beta3 semantics.  That would have caught the issue where the api version of the endpoint doesn't have to align to the internal Kube objects (for instance, if the kubelet API evolves independently, v1.2 may map to kube v1, and OpenShift definitely will rev at different time periods).